.github/workflows/ci.yaml

```diff
@@ -48,12 +50,17 @@ jobs:
         run: cargo build --verbose
       - name: Cargo Check
         run: cargo check
+      - name: Test Setup
+        run: cargo run --example test-setup
```

Comment:
The setup and teardown binaries are defined as examples in Cargo.toml. It is hacky to do that, but normal bins don't have access to dev dependencies, so it would be hard to get access to tokio without locking that dependency behind a feature flag. There might be a better way to organize that, but I couldn't find one.

Reply:
nope that seems fine but we might want to put a comment here. i can do it.